File: greedy/1663_smallest_string_with_a_given_numeric_value/1663_smallest_string_with_a_given_numeric_value.go
package main

import "fmt"

func getSmallestString(n int, k int) string {
	t := k - n
	ans := make([]byte, n, n)
	for i := n - 1; i >= 0; i-- {
		if t > 25 {
			ans[i] = 'z'
			t -= 25
		} else {
			ans[i] = byte('a' + t)
			t = 0
		}
	}

	return string(ans)
}

func main() {
	n := 3
	k := 27
	res := getSmallestString(n, k)
	fmt.Println(res)
}

File: greedy/1663_smallest_string_with_a_given_numeric_value/1663_smallest_string_with_a_given_numeric_value.py
class Solution:
    def getSmallestString(self, n: int, k: int) -> str:
        k -= n
        cnt_z = k // 25
        last = k % 25
        return 'a' * (n - 1 - cnt_z) + chr(ord('a') + last) + 'z' * cnt_z


n = 3
k = 27
res = Solution().getSmallestString(n, k)
print(res)
